src/Calculator/Common/KeyboardShortcuManager.cs

```diff
@@ -0,0 +1,866 @@
+            {
+                int viewId = Utilities.GetWindowId();
+
+                if (controlKeyPressed)
+                {
+                    if (altPressed)
+                    {
+                        return null;
+                    }
+                    else
+                    {
+                        if (shiftKeyPressed)
+                        {
+                            return s_VirtualKeyControlShiftChordsForButtons[viewId];
+                        }
+                        else
+                        {
+                            return s_VirtualKeyControlChordsForButtons[viewId];
+                        }
+                    }
+                }
+                else
+                {
+                    if (altPressed)
+                    {
+                        if (!shiftKeyPressed)
+                        {
+                            return s_VirtualKeyAltChordsForButtons[viewId];
+                        }
+                        else
+                        {
+                            return null;
+                        }
+                    }
+                    else
+                    {
+                        if (shiftKeyPressed)
+                        {
+                            return s_VirtualKeyShiftChordsForButtons[viewId];
+                        }
+                        else
+                        {
+                            return s_virtualKey[viewId];
+                        }
+                    }
+                }
+            }
+
+            // CSHARP_MIGRATION: TODO: double check below design
+            // EqualRange is a helper function to pick a range from std::multimap.
+            private static IEnumerable<TValue> EqualRange<TKey, TValue>(SortedDictionary<TKey, List<TValue>> source, TKey key)
+            {
+                Debug.Assert(source != null);
+                if(source.TryGetValue(key, out List<TValue> items))
+                {
+                    return items;
+                }
+                else
+                {
+                    return Enumerable.Empty<TValue>();
+                }
+            }
+
+            // CSHARP_MIGRATION: TODO: double check below design
+            // Insert is a helper function to insert a pair into std::multimap.
+            private static void Insert<Tkey, TValue>(SortedDictionary<Tkey, List<TValue>> dest, Tkey key, TValue value)
+            {
+                if(dest.TryGetValue(key, out List<TValue> items))
+                {
+                    items.Add(value);
+                }
+                else
+                {
+                    items = new List<TValue> { value };
+                    dest.Add(key, items);
+                }
+            }
+
+            private static SortedDictionary<int, SortedDictionary<char, List<WeakReference>>> s_characterForButtons = new SortedDictionary<int, SortedDictionary<char, List<WeakReference>>>();
```

Comment:
Enum MyVirtualKey from CalcViewModel is available now. Could you please convert char type back to MyVirtualKey?

Reply:
Fixed: changed those `char` types back to `MyVirtualKey`